File: #1~20/#1-两数之和/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> result_map;
        for(int i = 0;i<nums.size();i++){
            if (result_map.count(nums[i]))
				return {result_map[nums[i]],i };
            result_map[target-nums[i]] = i;
        }
        return {};
    }
};

File: 1~20/1-两数之和/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result_map = {}
        for index, num in enumerate(nums):
            #if num in result_map.keys():  时间很长
            if num in result_map:   #时间很短
                return( [result_map[num] , index] )
            else:
                result_map[target-num] = index
